### PR TITLE
[Search v2.1] Create In filter #46030

### PR DESCRIPTION
**Pull request by Guccio163**
### Details

This PR adds the `In` filter - part of `Search v2.1` milestone

### Fixed Issues

$ https://github.com/Expensify/App/issues/46030
PROPOSAL:

### Tests

1. Open `search/filters`, no options selected should show in `In` filter section
2. Open `In` filter page, all options should remain unchecked 
3. Check some options, they should propagate on the top of list with selected checkmark ✅ Also selected options should not be visible in default propositions (recent reports)
4. Try to search for an (non-selected) option in search bar; it should come up, search results should replace default propositions (recent reports) for the time of searching
5. Select searched option; selected option should be displayed with the rest on top of list, search phrase should reset, search results should be replaced back by default propositions (recent reports)
6. Search for the same option once again, it should come up but this time with selected checkmark ✅
7. Click on the selected option, it should uncheck and disappear from selected list; If it was previously in recent reports, it should pop up back there

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

N/A

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/b3a7a481-fcde-4c2f-97e4-04e935ba216d

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/12008787-9dfd-4281-8a7c-4bcc45eaa3bf

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/35b1cefa-32cf-49f0-8203-1561c7aac76b

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/23e1ff5b-8a2c-43b8-82e3-c3105fdb67ea

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/b5f6ba13-5ac8-4f07-8ae9-92049fb26ad3

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/06538284-984b-4170-a0fd-10e316b081e5

</details>

### CODE REVIEW

File: src/components/Search/SearchFiltersChatsSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    const [searchTerm, debouncedSearchTerm, setSearchTerm] = useDebouncedState('');
+    const cleanSearchTerm = useMemo(() => searchTerm.trim().toLowerCase(), [searchTerm]);
+
+    const defaultOptions = useMemo(() => {
+        if (!areOptionsInitialized || !isScreenTransitionEnd) {
+            return defaultListOptions;
+        }
+        const optionList = OptionsListUtils.getSearchOptions(options, '');
+        const header = OptionsListUtils.getHeaderMessage(optionList.recentReports.length + optionList.personalDetails.length !== 0, !!optionList.userToInvite, '');
+        return {...optionList, headerMessage: header};
```
Comment (Kicu) [inline src/components/Search/SearchFiltersChatsSelector.tsx:62]: I believe this `headerMessage` here might be not needed. Do you know if it affects anything?
If not remove it and return `optionList` directly.

**Review by Kicu — COMMENTED** [review]
Quick review about 1 thing

File: src/components/Search/SearchFiltersChatsSelector.tsx (hunk 1 above)
Comment (Guccio163) [reply src/components/Search/SearchFiltersChatsSelector.tsx:62]: Good catch, the headerMessage is used here but in opposite to `chat-finder`, it is set together with `sections`, this part is useless then 👍

**Review by Kicu — APPROVED** [review]
very good job with this quite problematic filter 😅 
I think the final code looks pretty good, I left several comments but most are minor name and codestyle changes

File: src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersInPage.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    return (
+        <ScreenWrapper
+            testID={SearchFiltersInPage.displayName}
+            includeSafeAreaPaddingBottom={false}
+            shouldShowOfflineIndicatorInWideScreen
+            offlineIndicatorStyle={styles.mtAuto}
+            onEntryTransitionEnd={handleScreenTransitionEnd}
+            shouldEnableMaxHeight
+        >
+            <HeaderWithBackButton title={translate('common.in')} />
```
Comment (Kicu) [inline src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersInPage.tsx:31]: Please add another prop here with a specific goBack like this:
```
                onBackButtonPress={() => {
                    Navigation.goBack(ROUTES.SEARCH_ADVANCED_FILTERS);
                }}
```.

Also if you can take a look at the rest of pages and add it to any that are missing. I think I forgot about this in some other filters as well

**Review by Kicu — CHANGES_REQUESTED** [review]
One small fix

File: src/components/Search/SearchFiltersChatsSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+            selectedOptions,
+            chatOptions.recentReports,
+            chatOptions.personalDetails,
+            personalDetails,
+            false,
+        );
+
+        newSections.push(formattedResults.section);
+
+        const visibleReportWhenSearching = chatOptions.recentReports.map((report) => (selectedReportIDs.includes(report.reportID) ? getSelectedOptionData(report) : report));
```
[the remark this answers is not in the document]
Comment (Guccio163) [reply src/components/Search/SearchFiltersChatsSelector.tsx:95]: Yeah, I was thinking about these names and wished you'd have a look at them, thanks for the idea!

File: src/components/Search/SearchFiltersChatsSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        const visibleReportsWithEmptySearch = chatOptions.recentReports.filter((report) => !selectedReportIDs.includes(report.reportID));
+        const recentReportsFiltered = cleanSearchTerm === '' ? visibleReportsWithEmptySearch : visibleReportWhenSearching;
```
[the remark this answers is not in the document]
Comment (Guccio163) [reply src/components/Search/SearchFiltersChatsSelector.tsx:97]: Right, in this usage we don't really need the `recent` keyword anymore, since all visible options are from `chatOptions.recentReports`.

File: src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersInPage.tsx (hunk 2 above)
Comment (Guccio163) [reply src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersInPage.tsx]: All filters checked and fixed where needed 👌

**Comment by Guccio163** [comment]
All comments from review included and applied 👌

File: src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersInPage.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                        });
+                    }}
+                    initialReportIDs={searchAdvancedFiltersForm?.in ?? []}
+                />
+            </View>
+        </ScreenWrapper>
+    );
+}
+
+SearchFiltersInPage.displayName = 'SearchFiltersStatusPage';
```
Comment (Kicu) [inline src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersInPage.tsx:54]: fix 
```suggestion
SearchFiltersInPage.displayName = 'SearchFiltersInPage';
```

File: src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersInPage.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            shouldShowOfflineIndicatorInWideScreen
+            offlineIndicatorStyle={styles.mtAuto}
+            onEntryTransitionEnd={handleScreenTransitionEnd}
+            shouldEnableMaxHeight
+        >
+            <HeaderWithBackButton title={translate('common.in')} />
+            <View style={[styles.flex1]}>
+                <SearchFiltersChatsSelector
+                    isScreenTransitionEnd={isScreenTransitionEnd}
+                    onFiltersUpdate={(initialIDs) => {
```
[the remark this answers is not in the document]
Comment (Kicu) [reply src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersInPage.tsx:42]: I think you forgot to change the name in here.

In here specifically I wanted this argument name changed, because this is a function that keeps updating filterIds in form state. Because of that `initial` part of the name in here is especially misleading.

**Review by Kicu — APPROVED** [review]
Hey I found 2 more renames to fix, but other than that the PR looks good.

I'm leaving an approve and feel free to open the PR to external review, just fix these 2 small renames first.

File: src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersInPage.tsx (hunk 5 above)
Comment (Guccio163) [reply src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersInPage.tsx]: Hawkeye, I overlooked it! 🎯

**Comment by Guccio163** [comment]
Hi @shawnborton, could you take a look if everything is ok from design standpoint?

**Comment by shawnborton** [comment]
Can you show me what the Search page looks like once this filter is applied? Thanks!

**Comment by Guccio163** [comment]
Yeah sure, right now the page after you find some matches looks like this:

https://github.com/user-attachments/assets/4631c9da-8158-4c0f-b433-fdc7c279b600

When now, it's just typical Empty page view.

**Comment by shawnborton** [comment]
Interesting, why are we just showing numbers for the `In:` in the page header?
CleanShot 2024-08-28 at 08 18 44@2x

**Comment by Guccio163** [comment]
Great catch, we pass them as IDs but I think it should display names; I'll change it in a white 👌

**Comment by Kicu** [comment]
FYI after reading this I have realised we do not handle this for multiple other filters as well. For example `From` and `To` will also display user id instead of their name/email.

Here's a bad looking  scenario:
Screenshot 2024-08-28 at 15 07 30
 
Because this is a more general problem, I would suggest to move handling of this (let's treat it as a bug) to a separate issue. We would like to move forward with this PR for now because it has been open for a long time, and someone else can handle pretty-printing these values as a follow-up

CC @luacmartins @Guccio163 @shawnborton

**Comment by Guccio163** [comment]
Thanks for that @Kicu, I think we should leave it to the separate issue too!

**Review by rayane-d — COMMENTED** [review]
<details>
<summary>Video</summary>

https://github.com/user-attachments/assets/c0348585-a4dd-422c-a378-2c5ec932889a

</details>

**Comment by Guccio163** [comment]
> Bug 2: All available reports options are included in the list. but I think we should include IOU reports options only
Are we sure? I mean it sound good, but doc stated that `In` should display the same data as `chat-finder`. We already discarded displaying accounts (unlike existing priv chats)

**Comment by Guccio163** [comment]
Interesting, it might end like this. `In` filter is described quite poorly in the doc, I figured it is supposed to look for f.ex. expenses made in specific reports, but I haven't confirmed it completely. @luacmartins, maybe you could shed some light on this?

**Comment by Guccio163** [comment]
This is everything I found in the doc on this topic:

Search v2.1 on Advanced Filters:
>The following filters will allow multi-select values:
(...)
In - list of chats found in the ChatFinderPage
(...)

Chats filter results:
>Search results will feature an In %reportName% label at the top of the result row.

**Comment by JmillsExpensify** [comment]
The `in` filter is search query syntax is intended to focus on `chat` type reports. Here's an excerpt from the design doc. 

CleanShot 2024-08-29 at 13 21 56@2x

Does that help?

**Comment by Guccio163** [comment]
[bug2] I don't agree with calling current displayed options a bug; Currently displayed are all options that are available in `ChatFinderPage` with exception to accounts;
tl;dr of the thread from Slack about this issue: We shouldn't display users because `ChatFinderPage` can both navigate to existing chat and create a new one if we doesn't have any with selected user. This filter should be able only to filter from existing reports (but not users, since they are accounts not reports) and hence we include all options used in `ChatFinderPage` except users (not to confuse with priv chats, we include them)

[bug1] I see it too, I'll check if there is some quick solution for it

I'm not sure if we should wait with this filter for #47918, since it shouldn't change anything in frontend usage

**Comment by Guccio163** [comment]
bug1 just been fixed, so I think it's ready for a review 👀

https://github.com/user-attachments/assets/1177fa7a-3b49-43a9-ad25-d00d6b1a6ac8

**Comment by Guccio163** [comment]
> Unread recent reports are displayed with bold text. But this is also the behaviour on "start chat" and "submit expense" flows, so I think this is expected.

Yeah, it works like that in `chat-finder` too and `In` should base on it so I think it's by design

> Bug on Android Native App: The display title is truncated even though there is available space. Same thing with the dropdown button title:

This is bigger than this filter (funny that no-one caught it in previous filters, huh), it works like that with most of (if not all) filters: 
Screenshot 2024-08-30 at 17 44 48

I really want to close and merge this issue, because it took a lot to make it work, but if it won't be merged on Monday I can take a look if I can fix it for all filters right away.

**Review by rayane-d — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Comment by Guccio163** [comment]
> I think we can merge this as is and address the UI issues in a follow up.

I'm all for it, we're not working until the end of the week, but I'll take that UI follow-up first thing on monday ⚡️

**Comment by luacmartins** [comment]
@Guccio163 conflicts

**Review by luacmartins — DISMISSED** [review]
LGTM

**Comment by Guccio163** [comment]
Godspeed ⚡️ Ready to merge 👌

**Comment by Guccio163** [comment]
Tests look like they're all about to pass; I couldn't find in which report type filter `In` should be visible, but from what I remember only in `Chats` - that's where it's added ✅

**Review by luacmartins — APPROVED** [review]

**Comment by luacmartins** [comment]
Thanks for pushing this to the finish line @Guccio163 @rayane-djouah

**Comment by mvtglobally** [comment]
@luacmartins Steps say NO QA, but seems like we can QA it. Can you pls confirm if we need to run the tests

File: src/components/Search/SearchFiltersChatsSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+            success
+            text={translate('common.save')}
+            pressOnEnter
+            onPress={() => {
+                onFiltersUpdate(selectedReportIDs);
+                Navigation.goBack(ROUTES.SEARCH_ADVANCED_FILTERS);
+            }}
+            large
+        />
+    );
```
Comment (ishpaul777) [inline src/components/Search/SearchFiltersChatsSelector.tsx:160]: {bz checklist} coming from https://github.com/Expensify/App/issues/70550, Reset button is not included in footer component here, while its included in other screens we added it in https://github.com/Expensify/App/pull/70869
